doc/sphinx/proof-engine/ltac.rst

```diff
@@ -451,178 +461,185 @@ structure:
 .. tacn:: @ltac_expr__1 || @ltac_expr__2
    :name: || (left-biased branching)
 
-   :n:`@ltac_expr__1` and :n:`@ltac_expr__2` are evaluated respectively to :n:`v__1` and
-   :n:`v__2` which must be tactic values. The tactic value :n:`v__1` is
+   :n:`@ltac_expr__1` and :n:`@ltac_expr__2` are evaluated to :n:`v__1` and
+   :n:`v__2` respectively, which must be tactic values. The tactic value :n:`v__1` is
    applied in each subgoal independently and if it fails *to progress* then
    :n:`v__2` is applied. :n:`@ltac_expr__1 || @ltac_expr__2` is
    equivalent to :n:`first [ progress @ltac_expr__1 | @ltac_expr__2 ]` (except that
    if it fails, it fails like :n:`v__2`). Branching is left-associative.
 
-Generalized biased branching
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-
-The tactic
+Generalized biased branching: tryif
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-.. tacn:: tryif @ltac_expr__1 then @ltac_expr__2 else @ltac_expr__3
+.. tacn:: tryif @ltac_expr__if then @ltac_expr__then else @ltac_expr2__else
    :name: tryif
 
    is a generalization of the biased-branching tactics above. The
-   expression :n:`@ltac_expr__1` is evaluated to :n:`v__1`, which is then
-   applied to each subgoal independently. For each goal where :n:`v__1`
-   succeeds at least once, :n:`@ltac_expr__2` is evaluated to :n:`v__2` which
-   is then applied collectively to the generated subgoals. The :n:`v__2`
-   tactic can trigger backtracking points in :n:`v__1`: where :n:`v__1`
+   expression :n:`@ltac_expr__if` is evaluated to :n:`v__if`, which is then
+   applied to each subgoal independently. For each goal where :n:`v__if`
+   succeeds at least once, :n:`@ltac_expr__then` is evaluated to :n:`v__then` which
+   is then applied collectively to the generated subgoals. The :n:`v__then`
+   tactic can trigger backtracking points in :n:`v__if`: where :n:`v__if`
    succeeds at least once,
-   :n:`tryif @ltac_expr__1 then @ltac_expr__2 else @ltac_expr__3` is equivalent to
-   :n:`v__1; v__2`. In each of the goals where :n:`v__1` does not succeed at least
-   once, :n:`@ltac_expr__3` is evaluated in :n:`v__3` which is is then applied to the
+   :n:`tryif @ltac_expr__if then @ltac_expr__then else @ltac_expr__else` is equivalent to
+   :n:`v__if; v__then`. In each of the goals where :n:`v__if` does not succeed at least
+   once, :n:`@ltac_expr__else` is evaluated in :n:`v__else` which is is then applied to the
    goal.
 
+   :tacn:`tryif` is an :token:`ltac_expr2`.
+
 Soft cut
 ~~~~~~~~
 
+.. todo PR Would like a different subsection title above, but I have trouble distinguishing
+   once and exactly_once.
```

Comment:
```coq
Goal True \/ False.
(* The following tactic has two successes but we stop at the first one. *)
right + left. (* we end up having to prove False *)
Restart.
(* This time we complete the proof because easy makes the first success
 * of [right + left] end up in a failure so that we try the second one. *)
right + left; easy.
Restart.
(* The following tactic has only one success. *)
once (right + left).
Restart.
(* Therefore, this does not work anymore. *)
Fail once (right + left); easy.
(* Since [right + left] has two successes, the following fails. *)
Fail exactly_once (right + left).
```

So the two have a quite different goal (one is to prevent the backtracking---cut---, the other is to check the behavior of the tactic---but to do that you need to explore all backtracking points: much more expensive!).

Reply:
The chapter needs an explanation of backtracking.  Maybe that's only a sentence or two.  I'll leave the todo for later.